decayAmplitude/waveSetGenerator.cc

```diff
@@ -565,6 +584,62 @@ waveSetGenerator::createNewDecayTopology(const isobarDecayTopology&  parentDecay
 }
 
 
+vector<isobarDecayTopologyPtr>
+waveSetGenerator::setMassDepsForDecayTopology(const isobarDecayTopologyPtr         decayTopology,
+                                              const particleProperties&            isobar,
+                                              const particleProperties::decayMode& decayMode)
+{
+	// set mass dependencies of isobar decay
+	set<string> usedMassDependencies;
+	vector<isobarDecayTopologyPtr> decayMassDeps;
+	// 1. search the decay modes of the current isobar for a match with the current decay
+	//    (there might be multiple matches (e.g. a special mass depedence for a
+	//    particular L))
+	for (size_t iDecay = 0; iDecay < isobar.nmbDecays(); ++iDecay) {
+		if (not (isobar.decayModes()[iDecay] == decayMode))
+			continue;
+
+		//  2. loop over mass dependencies
+		const vector<string>& massDeps = isobar.decayModes()[iDecay]._massDependencies;
+		if (massDeps.size() != 0) {
+			for (vector<string>::const_iterator massDepType = massDeps.begin(); massDepType != massDeps.end(); ++massDepType) {
+				if (usedMassDependencies.count(*massDepType) != 0)
+					continue;
+				usedMassDependencies.insert(*massDepType);
+				if (*massDepType == relativisticBreitWigner::Name()) usedMassDependencies.insert("");
+				if (*massDepType == "") usedMassDependencies.insert(relativisticBreitWigner::Name());
+
+				massDependencePtr massDep = createMassDependence(*massDepType);
```

Comment:
variable not needed anywhere else; maybe move createMassDependence() directly into if statement below